### ISSUE TITLE
Documentation does not invite to contribution

### ISSUE THREAD

**Issue by fwolfst** [dry-rb/dry-rb.org#270]
As I stumbled upon a mistake (I believe) in the documentation I wanted to contribute to it. I was surprised how long it took me to find out where it is and what I would have to do (after searching the repos, the discussions, I asked in the chat).

I propose that the online-documentation-site itself invites to documentation improvements (with a link to the repo). Besides, every repositories README should - in my view - link to the website-repository and explanations about how to contribute to the documentation.

Also, I dont want to rule out the possibility that I was a bit blind, in that case rephrase it to "doesnt invite ENOUGH to contribution ...'.

Besides that I would prefer to have each gems documentation in its own repo and distributed with the gem. This has multiple advantages (e.g. offline-capability, documentation in sync with the used version, invitation to improvements; https://github.com/fwolfst/your-repository/issues/2 and https://github.com/fwolfst/your-repository/issues/1), but obviously makes the current rendering process somewhat more complicated. If this (second) issue shall be adressed, I am happy to create a second issue for it.

**Comment by Morozzzko** [comment]
I was actually thinking about splitting the docs into separate repositories. However, it makes the documentation bundling so complicated that I gave up.

It would be great if you could help me answer some of the questions.

Let's say we split `dry-rb.org` and put tutorials & docs for each gem into according `dry-rb/#{gem}` repo. 

* Let's say we only have `dry-monads` locally. Should we be able to build them? How is it going to look? 
* How do we build `dry-rb.org`? Should we use `git submodule` / `git subtree` to fetch the docs? Perhaps, something else?

**Comment by fwolfst** [comment]
> Let's say we only have dry-monads locally. Should we be able to build them? How is it going to look?

Following the example, I'd put the stuff from https://github.com/dry-rb/dry-rb.org/tree/master/source/gems/dry-monads into `dry-rb/dry-monads/docs` (without the partials, images and other assets). Who wants to read the documentation offline should be fine with reading the markdown files, imho - I find them pretty readable and very helpful.

> How do we build dry-rb.org? Should we use git submodule / git subtree to fetch the docs? Perhaps, something else?

About how the original https://dry-rb.org/ is built i cannot say much (except that its middleman). Either pull the latest tags during the build process (via the/a Rakefile (https://github.com/avillafiorita/middleman-rakefile/blob/master/Rakefile) and some middleman-config?) or use git submodules; about which I do not know much and more often than not had to struggle with. I guess the middleman-devs and users could explain how its done the least painful.

However its done, if not carefully crafted (and documented) afaiu the writers could lose the possibility of "live rendering" in development mode of middleman (or maybe that can be overcome with the submodules), as the workflow would probably include committing the changes before they can be seen; but again, probably submodules would actually make it possible to first change, confirm visually and then commit.

**Comment by Morozzzko** [comment]
Could you please build a proof-of-concept of your idea, or probably draw a diagram of your idea?

I tried to figure out how to use the docs using git submodules, but it feels like it's going to be unreasonably complex. 


Let's say I want to add new monads to the `dry-monads` gem. I'm going to add some extra docs and tutorials, so I want to see it live. I need to fork & clone two repos: `dry-monads` and `dry-rb.org`. When I'm done, I have to send two pull requests to the corresponding repos. Slow, but alright.

Now, let's say we store docs in `dry-monads/docs` and use **git submodules**. Then, I'll have to follow this path:

* Fork & clone `dry-rb.org`
* Open local `dry-rb.org` repo, navigate to `dry-monads` submodule and change its remote to match your fork
* When I'm done, I send a pull request to `dry-monads`
* Once it's merged, I ask someone to pull the latest changes into `dry-rb.org` repo

It seems like the simplest path out there. However, if you want to store `dry-monads` in a separate directory, it's gonna be a painful experience.


So, I was wondering: how can we make the experience less painful?

Perhaps, we *don't really need* to store docs in separate repositories. I think a simple section in a README would be enough to make it easier to contribute.

**Comment by jimcavoli** [comment]
FWIW, every additional step to be able to contribute is another point on the funnel where someone becomes more likely to just drop off and decide it isn't worth it. I think for the sake of trying to increase the head count and participation rate of the community, it would be advantageous to keep the docs here and try to be better about having contributors send some additional PR this way when something rather substantial changes. We should consider whose use case(s) we're optimizing for the most. I'd argue that most code contributors are generally more prepared for extra or optional steps, so it would make more sense to have them run an extra command or repo to keep track of for "also docs" without making things too difficult for others who could add useful examples, copy edit, and so on all from one place. A lot of those types of contributors may well also be users of many gems, but not able to take all the time required for fixing a bug or something, and shouldn't have a harder time enhancing the content for as much as they're prepared to.

That's my opinion, anyway. Happy to discuss those points further now and in the future and reconsider the conclusion. These would be the key attributes and roles I'd be most sensitive to analyzing in a proposed changeup of documentation system though.

**Comment by JoshTGreenwood** [comment]
FWIW I stumbled upon a typo and it took me a few minutes to find this repository. I would have loved a "fork this on Github" button where I could jump straight to the file in question.

**Comment by Morozzzko** [comment]
The discussion has been idle for a while, as the topic is too huge to grasp.

@JoshTGreenwood thank you for the suggestion, I addressed it in #342. It's not a fancy `fork me` button, but a simple link in the footer. But I think it's still better than nothing 🎉 

I hope we'll manage to get a more elaborate solution when we switch documentation engine. There's some discussion about it in [Zullip](https://dry-rb.zulipchat.com/#narrow/stream/191662-general/topic/changing.20documentation.20hosting)

**Comment by solnic** [comment]
Just a quick update that I'll be extracting docs from this repo to individual gem repos **soon**. I already upgraded the site to use middleman 5 + built middleman-docsite extension that is used by dry-rb.org and rom-rb.org. Extracting gem docs is the last step of this whole upgrade/migration process.

**Comment by solnic** [comment]
We now have "Edit on GitHub" buttons on all gem pages 😄 Closing via #365

**Comment by fwolfst** [comment]
Awesome!
